includes/class-wc-data-structurer.php

```diff
@@ -0,0 +1,164 @@
+      }
+    }
+    else {
+      $structured_data = $this->structured_data[0];
+    }
+    return wp_json_encode( $context + $structured_data );
+  }
+
+  /**
+   * Contructor
+   */
+  public function __construct() {
+    add_action( 'woocommerce_before_shop_loop_item', array( $this, 'init_product_category_structured_data' ) );
+    add_action( 'woocommerce_single_product_summary', array( $this, 'init_product_structured_data' ) );
+    add_action( 'woocommerce_review_meta', array( $this, 'init_product_review_structured_data' ) );
+    add_action( 'wp_footer', array( $this, 'enqueue_structured_data' ) );
+  }
+
+  /**
+   * If structured data is set, echoes the encoded structured data into the `wp_footer` action hook.
+   */
+  public function enqueue_structured_data() {
+    if ( $structured_data = $this->get_structured_data() ) {
+      echo '<script type="application/ld+json">' . $structured_data . '</script>';
+    }
+  }
+
+  /**
+   * Generates the product category structured data...
+   * Hooked into the `woocommerce_before_shop_loop_item` action hook...
+   */
+  public function init_product_category_structured_data() {
+    if ( ! is_product_category() ) {
+      return;
+    }
+    $this->init_product_structured_data();
+  }
+  /**
+   * Generates the product structured data...
+   * Hooked into the `woocommerce_single_product_summary` action hook...
+   * Applies the `woocommerce_product_structured_data` filter hook for clean structured data customization...
+   */
+  public function init_product_structured_data() {
+    global $product;
+
+    $json['@type']             = 'Product';
+    $json['@id']               = 'product-' . get_the_ID();
+    $json['name']              = get_the_title();
+    $json['image']             = wp_get_attachment_url( $product->get_image_id() );
+    $json['description']       = get_the_excerpt();
+    $json['url']               = get_the_permalink();
+    $json['sku']               = $product->get_sku();
+    $json['brand']             = array(
+      '@type'                  => 'Thing',
+      'name'                   => $product->get_attribute( __( 'brand', 'woocommerce' ) )
+    );
+    if ( $product->get_rating_count() ) {
+      $json['aggregateRating'] = array(
+        '@type'                => 'AggregateRating',
+        'ratingValue'          => $product->get_average_rating(),
+        'ratingCount'          => $product->get_rating_count(),
+        'reviewCount'          => $product->get_review_count()
+      );
+    }
+    $json['offers']            = array(
+      '@type'                  => 'Offer',
+      'priceCurrency'          => get_woocommerce_currency(),
+      'price'                  => $product->get_price(),
+      'itemCondition'          => 'http://schema.org/NewCondition',
+      'availability'           => 'http://schema.org/' . $stock = ( $product->is_in_stock() ? 'InStock' : 'OutOfStock' ),
+      'seller'                 => array(
```

Comment:
Not sure this should be included in a standard install - again products may be external, so the seller isn't necessarily correct.